//! new-transaction
import 0x0.VDF;
import 0x0.Debug;
main() {
  let challenge: vector<u8>;
  let difficulty: u64;
  let false_solution: vector<u8>;
  let true_solution: vector<u8>;
  let re: bool;

  difficulty = 100;
  challenge = h"aa";
  false_solution = h"005271e8f9"; // this needs to be an even number verify will fail otherwise. BTW need to catch that error below.

  re = VDF.verify(move(challenge), move(difficulty), move(false_solution)); // need to catch errors here.
  assert(move(re)==false, 7);

// this string is a valid proof, should solve to true
  true_solution = h"005271e8f9ab2eb8a2906e851dfcb5542e4173f016b85e29d481a108dc82ed3b3f97937b7aa824801138d1771dea8dae2f6397e76a80613afda30f2c30a34b040baaafe76d5707d68689193e5d211833b372a6a4591abb88e2e7f2f5a5ec818b5707b86b8b2c495ca1581c179168509e3593f9a16879620a4dc4e907df452e8dd0ffc4f199825f54ec70472cc061f22eb54c48d6aa5af3ea375a392ac77294e2d955dde1d102ae2ace494293492d31cff21944a8bcb4608993065c9a00292e8d3f4604e7465b4eeefb494f5bea102db343bb61c5a15c7bdf288206885c130fa1f2d86bf5e4634fdc4216bc16ef7dac970b0ee46d69416f9a9acee651d158ac64915b000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001"

  re = VDF.verify(move(challenge), move(difficulty), move(true_solution)); // need to catch errors here.
  assert(move(re)==false, 7);

  return;
}
